include/zephyr/drivers/flash.h

```diff
@@ -249,6 +256,27 @@ static inline int z_impl_flash_erase(const struct device *dev, off_t offset,
 	return rc;
 }
 
+#if defined(CONFIG_FLASH_RESET)
+/*
+ *  @brief  Send SPI reset command.
+ *
+ *  This api can be used  to reset the SPI flash memory i.e volatile
+ *  register configurations before SPI access.
+ *  Command also aborts any ongoing internal SPI operations.
+ *
+ *  @param  dev             : flash dev
+ */
+__syscall void flash_reset(const struct device *dev);
+
+static inline void z_impl_flash_reset(const struct device *dev)
```

Comment:
You also need to provide a `z_vrfy_flash_reset` implementation in flash_handlers.c

Reply:
i have changed the implementation based on @de-nordic comments. Kindly review the new one.